src/main/java/teammates/ui/controller/AdminSessionsPageAction.java

```diff
@@ -146,17 +142,18 @@ private ActionResult createShowPageResultIfParametersInvalid() {
             statusToAdmin = "Admin Sessions Page Load<br>"
                           + "<span class=\"bold\"> Error: Missing Parameters</span>";
 
-            prepareDefaultPageData(calStart, calEnd);
+            prepareDefaultPageData(TimeHelper.convertLocalDateTimeToInstant(start, zone),
+                    TimeHelper.convertLocalDateTimeToInstant(end, zone));
             data.init(this.map, this.sessionToInstructorIdMap, this.totalOngoingSessions,
                       this.totalOpenStatusSessions, this.totalClosedStatusSessions, this.totalWaitToOpenStatusSessions,
-                      this.totalInstitutes, this.rangeStart, this.rangeEnd, this.zone, this.isShowAll);
+                      this.totalInstitutes, this.rangeStart, this.rangeEnd, this.timeZone, this.isShowAll);
             return createShowPageResult(Const.ViewURIs.ADMIN_SESSIONS, data);
 
         }
 
-        this.rangeStart = start;
-        this.rangeEnd = end;
-        this.zone = zone;
+        this.rangeStart = TimeHelper.convertLocalDateTimeToInstant(start, zone);
+        this.rangeEnd = TimeHelper.convertLocalDateTimeToInstant(end, zone);
```

Comment:
The logic in `AdminSessionPageData` all requires local time. Maybe it would be better to save the local times to the instance fields and only convert it to instant when querying the database? Then we don't need to convert from instant to local time again in page data.